app/scripts/lib/rpc-method-middleware/handlers/wallet-revokePermissions.test.ts

```diff
@@ -0,0 +1,213 @@
+    };
+    handler(emptyRequest);
+    expect(end).toHaveBeenCalledWith(
+      invalidParams({ data: { request: emptyRequest } }),
+    );
+  });
+
+  // @ts-expect-error This is missing from the Mocha type definitions
+  describe.each([
+    [RestrictedMethods.eth_accounts],
+    [PermissionNames.permittedChains],
+  ])('%s permission is specified', (permission: string) => {
+    it('gets permissions for the origin', () => {
+      const { handler, getPermissionsForOrigin } = createMockedHandler();
+
+      handler({
+        ...baseRequest,
+        params: [
+          {
+            [permission]: {},
+          },
+        ],
+      });
+      expect(getPermissionsForOrigin).toHaveBeenCalled();
+    });
+
+    it('revokes the CAIP-25 endowment permission', () => {
+      const { handler, revokePermissionsForOrigin } = createMockedHandler();
+
+      handler({
+        ...baseRequest,
+        params: [
+          {
+            [permission]: {},
+          },
+        ],
+      });
+      expect(revokePermissionsForOrigin).toHaveBeenCalledWith([
+        Caip25EndowmentPermissionName,
+      ]);
+    });
+
+    it('revokes other permissions specified', () => {
+      const { handler, revokePermissionsForOrigin } = createMockedHandler();
+
+      handler({
+        ...baseRequest,
+        params: [
+          {
+            [permission]: {},
+            otherPermission: {},
+          },
+        ],
+      });
+      expect(revokePermissionsForOrigin).toHaveBeenCalledWith([
+        'otherPermission',
+        Caip25EndowmentPermissionName,
+      ]);
+    });
+
+    it('throws an error when a CAIP-25 permission exists from the multichain flow (isMultichainOrigin: true)', () => {
+      const { handler, getPermissionsForOrigin, end } = createMockedHandler();
+      getPermissionsForOrigin.mockReturnValue({
+        [Caip25EndowmentPermissionName]: {
+          id: '1',
+          parentCapability: Caip25EndowmentPermissionName,
+          caveats: [
+            {
+              type: Caip25CaveatType,
+              value: {
+                requiredScopes: {},
+                optionalScopes: {},
+                isMultichainOrigin: true,
+              },
+            },
+          ],
+        },
+      });
+
+      handler({
+        ...baseRequest,
+        params: [
+          {
+            [permission]: {},
+            otherPermission: {},
+          },
+        ],
+      });
+      expect(end).toHaveBeenCalledWith(
+        new Error(
+          'Cannot modify permission granted via the Multichain API. Either modify the permission using the Multichain API or revoke permissions and request again.',
+        ),
+      );
+    });
+  });
+
+  it('revokes permissions other than eth_accounts, permittedChains, CAIP-25 if specified', () => {
+    const { handler, revokePermissionsForOrigin } = createMockedHandler();
+
+    handler(baseRequest);
+    expect(revokePermissionsForOrigin).toHaveBeenCalledWith([
+      'otherPermission',
```

Comment:
Where is `otherPermission` specified?

If this is being set up in `createMockedHandler` what are your thoughts on making it explicit here?

Reply:
https://github.com/MetaMask/metamask-extension/pull/27847/commits/3442d7dc85aaae0a52e67f683c4b49ace4ebf702